### ISSUE TITLE
node-gyp error while installing oracledb module on AIX6.1

### ISSUE THREAD

**Issue by annjawn** [nodejs/node-gyp#1166]
I am trying to install `oracledb` on AIX6.1 using 

`npm install -g oracledb`

I am getting the below error 

```
.....
In file included from /home/oretail/.node-gyp/6.10.0/include/node/node.h:42:0,
                 from ../src/njs/src/njsOracle.cpp:51:
/home/oretail/.node-gyp/6.10.0/include/node/v8.h:21:19: fatal error: utility: No such file or directory
 #include <utility>
                   ^
compilation terminated.
oracledb.target.mk:123: recipe for target 'Release/obj.target/oracledb/src/njs/src/njsOracle.o' failed
gmake: *** [Release/obj.target/oracledb/src/njs/src/njsOracle.o] Error 1
gmake: Leaving directory '/home/oretail/node/lib/node_modules/oracledb/build'
gyp ERR! build error 
gyp ERR! stack Error: `gmake` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/oretail/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System AIX 6.1
gyp ERR! command "/home/oretail/node/bin/node" "/home/oretail/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/oretail/node/lib/node_modules/oracledb
gyp ERR! node -v v6.10.0
gyp ERR! node-gyp -v v3.5.0
gyp ERR! not ok 
...
```

I can see the `v8.h` file in the `.node-gyp` directory. I also have C/C++ compiler and gmake installed in the server and below is the `g++` version. Am I facing a compatibility issue here? 

```
g++ --version
g++ (GCC) 4.8.3
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by richardlau** [comment]
cc @nodejs/platform-aix

**Comment by bnoordhuis** [comment]
`<utility>` is a system header.  I don't know how you installed g++ but you probably also need to install the accompanying devel headers.

At any rate, it's neither a node-gyp nor an oracledb issue but a toolchain issue local to your system.

**Comment by annjawn** [comment]
I installed gcc/g++ using the rpm for AIX6.1 using `gcc-c++-4.8.3-1.aix6.1.ppc.rpm` , I also have symlink for `libstdc++.a` to `-> ../lib/gcc/powerpc-ibm-aix6.1.0.0/4.8.3/ppc64/libstdc++.a` on the server and `libgcc_s.a -> ../lib/gcc/powerpc-ibm-aix6.1.0.0/4.8.3/ppc64/libgcc_s.a`.

**Comment by annjawn** [comment]
Ok i think the `devel` headers are missing we only have the standard c++ libraries installed. Thanks.

**Comment by gibfahn** [comment]
FYI, you can see how we setup the Node.js community CI machines [here](https://github.com/nodejs/build/blob/master/setup/aix61/manualBootstrap.md#gcc).

**Comment by annjawn** [comment]
@gibfahn thank you for that. In my case the `devel` library was missing. As @bnoordhuis mentioned, I went ahead and checked and then installed the same following which I was able to install `oracledb` without any issues. I faced some other issues trying to make the installed `oracledb` module especially a step where the unix admin suggested running `mkdev -l iocp0` which magically solved the problem. I am not entirely sure what the issue was but it looked like there was some port related issue with the instant client installation files.
